nevergrad/parametrization/core.py

```diff
@@ -0,0 +1,497 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import uuid
+import warnings
+from collections import OrderedDict
+import typing as t
+import numpy as np
+
+
+BP = t.TypeVar("BP", bound="BaseParameter")
+P = t.TypeVar("P", bound="Parameter")
+D = t.TypeVar("D", bound="Dict")
+
+
+class Descriptors(t.NamedTuple):
+    """Provides access to a set of descriptors for the parametrization
+    This can be used within optimizers.
+    """
+    deterministic: bool = True
+    continuous: bool = True
+
+
+class NotSupportedError(RuntimeError):
+    """This type of operation is not supported by the parameter.
+    """
+
+
+# pylint: disable=too-many-instance-attributes
+class BaseParameter:
+    """This provides the core functionality of a parameter, aka
+    value, subparameters, mutation, recombination
+    """
+
+    def __init__(self, **subparameters: t.Any) -> None:
+        self.uid = uuid.uuid4().hex
+        self.parents_uids: t.List[str] = []
+        self._subparameters = None if not subparameters else Dict(**subparameters)
+        self._dimension: t.Optional[int] = None
+
+    @property
+    def descriptors(self) -> Descriptors:
+        return Descriptors(deterministic=True, continuous=True)
```

Comment:
As `deterministic` and `continuous` are not related, why not have 2 `property` in the class directly?

Reply:
In current instrumentation we have 2 different properties, but little by little (like as of today) more different properties like this one are creeping in (like "noisy_function", "discrete" etc). I'd rather we keep such "descriptors" contained somewhere to ease the interface, but I'm not 100% sure either